### PR TITLE
feat: events: make event saving configurable independent of eth api

### PR DESCRIPTION
**Pull request by Stebalien**
## Related Issues

Related to https://github.com/filecoin-project/lotus/issues/9774

## Proposed Changes

We can now instruct lotus to store events, even if the Eth JSON-RPC API is disabled.

## Additional Info

Bikeshed on name?

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by Stebalien** [comment]
(I'll run `make gen` and add a test once we settle on the name and whether or not we actually want this)

File: node/config/types.go [hunk 1]
```diff
@@ -691,6 +691,10 @@ type FevmConfig struct {
 }
 
 type Events struct {
+	// Save specifies whether or not to save emitted events, or discard them. Events are
+	// discarded by default.
+	Save bool
```
Comment (magik6k) [inline node/config/types.go:696]: Name sounds good

We should make the comment say that it is enabled implicitly when EnableEthRPC is set - or we can make the node when save isn't set when EnableEthRPC is (can be done easily with the `Error` node builder option)

**Comment by magik6k** [comment]
> I'll run make gen

Note that you can also copy the output from the gen-check CI diff into `git apply`

**Comment by arajasek** [comment]
@Stebalien I have no problem with this, but would lean towards not merging this unless you can point me to someone that wants it? Lotus has a lot of knobs already, and FEVM added yet more (all necessary). I'm a little disinclined to add another finer-tuned one.

**Comment by Stebalien** [comment]
IIRC, @hsanjuan and friends wanted this for archival. Although I think they ended up just enabling the API. So yeah, I don't think it's critical.

**Comment by Stebalien** [comment]
@ribasushi want to submit a rebased PR with docs and tests?

**Pull request closed without merging**